jt-classbreaks/src/main/java/it/geosolutions/jaiext/classbreaks/NaturalBreaksHistogramOpImage.java

```diff
@@ -166,5 +169,25 @@ protected void postCalculate(Classification c, int band) {
         }
         breaks[0] = buckets.get(0).getMin();
         hc.setBreaks(band, breaks);
+        if(percentages.booleanValue())
+            hc.setPercentages(getPercentages(buckets, m, breaks));
+    }
+
+    private double[] getPercentages(List<Bucket> buckets, int m, Double[] breaks) {
+        double [] percentages = new double [numClasses];
+        double [] values = buckets.stream().mapToDouble(b->b.getMin()).toArray();
+        for (int i = 0; i< numClasses; i++){
+            double break1 = breaks[i];
+            double break2 = breaks[i+1];
+            boolean last = i==(numClasses-1);
+            double classMembers;
+            if (last){
+                classMembers= DoubleStream.of(values).filter(v->v>=break1 && v<=break2).count();
```

Comment:
Usage of streams normally provides lower performance compared to loops, please avoid them in high volume situations and use a classic array for loop. JAI development requires a C programmer mindset.

Reply:
fixed